packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -236,115 +64,45 @@ export function buildCategoryFacet(
     throw loadReducerError;
   }
 
-  const controller = buildController(engine);
+  const controller = buildCoreCategoryFacet(engine, props);
```

Comment:
Nitpick: A small change on the name since we are already in a `controller`, just for distinction

```suggestion
  const coreController = buildCoreCategoryFacet(engine, props);
```